### ISSUE TITLE
Batch Norm Update & Loss Avg for Grad accumulation

### ISSUE THREAD

**Issue by jaideep11061982** [NVIDIA/DeepLearningExamples#625]
1) I see in grad accumulatio implementation 
https://github.com/NVIDIA/DeepLearningExamples/blob/9e457013c07c6008585b0497a2d1d33a2eae2bcd/PyTorch/Segmentation/MaskRCNN/pytorch

We are dividing weights/number of steps  why is that done so 

2)  we are not dividing the loss in above case as done in grad accumulation of Speech/Language models.  Is it necessary step to divide the loss also 

3) Do we also need to adjust batch norm stats like running mean or so as per batch accumulation step ?

**Comment by sharathts** [comment]
We are dividing param.grad.data (gradients) not the weights.
It is equivalent to either divide the gradient or the loss.

3. Batchnorm stats get updated based on the stats from the current batch (not the accumulated batch). This setup converges fine.

**Comment by sharathts** [comment]
Closing issue. Feel free to reopen if you have more questions.
